### PR TITLE
Update underglow.md

### PR DESCRIPTION
**Pull request by Megamannen**
Updated the documentation according to my current understandring of how to pin-mapping works, also split the chapter into a nrf-section and a non-nrf-section since that information isn't applicable to both

### CODE REVIEW

**Review by mcrosson — APPROVED** [review]
This looks good

Thank you!

**Review by Nicell — CHANGES_REQUESTED** [review]
Thanks for adding this information. Could you also run the prettier and linting commands? You can find a guide [here](https://deploy-preview-709--zmk.netlify.app/docs/development/documentation#formatting-and-linting-your-changes).

File: docs/docs/features/underglow.md [hunk 1]
```diff
@@ -87,11 +92,14 @@ Here's an example of an nRF52 SPI definition:
 
 :::info
 
-If you are configuring SPI for an nRF52840 (or other nRF52) based board, double check that you are using pins that aren't restricted to low frequency I/O.
+If you are configuring SPI for an nRF52 based board, double check that you are using pins that aren't restricted to low frequency I/O.
 Ignoring these restrictions may result in poor wireless performance. You can find the list of low frequency I/O pins [here](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fps_nrf52840%2Fpin.html&cp=4_0_0_6_0).
```
[the remark this answers is not in the document]
Comment (Megamannen) [reply docs/docs/features/underglow.md]: Done!

**Review by Nicell — APPROVED** [review]
Thanks!

**Pull request merged**